test/specs/rules/require-var-decl-first.js

```diff
@@ -0,0 +1,94 @@
+                { statement: 'var a;\nfunction b()\n{\nvar c;\n}', errorCount: 0 },
+                { statement: 'var a;var b=function(){var c;};', errorCount: 0 },
+                { statement: 'var a;var b=function(){var c;/*block comment*/var e;};', errorCount: 0 },
+                {
+                    statement: 'var a;var b=function(){var c;/*block comment*//*block comment2*/var e;};',
+                    errorCount: 0
+                },
+                { statement: 'var a;/*block comment*//*block comment2*/var b;', errorCount: 0 },
+                { statement: 'var a;var count;for(count=0;count<10;count++){}', errorCount: 0 },
+                { statement: 'var a;var count;var b;while(count<10){b=1;}', errorCount: 0 },
+                { statement: 'var a;var count;var b;do{b=1;}while(count<10);', errorCount: 0 },
+                { statement: 'var a;a=1;var b;', errorCount: 1 },
+                { statement: '/* block comment */var a;a=1;var b;', errorCount: 1 },
+                { statement: 'var a;a=1;var b;/* block comment */', errorCount: 1 },
+                { statement: 'var a;/*block comment*/a=1;/*block comment2*/var b;', errorCount: 1 },
+                { statement: 'var a;function b(){var c;}var d;', errorCount: 1 },
+                { statement: 'var a;function b(){var c;}var d;', errorCount: 1 },
+                { statement: 'var a;function b(){var c;c=1;var d;}', errorCount: 1 },
+                { statement: 'var a;for(var count=0;count < 10;count++){}', errorCount: 1 },
+                { statement: 'var a;var count;while(count<10){var b=1;}', errorCount: 1 },
+                { statement: 'var a;var count;do{var b= 1;}while(count<10);', errorCount: 1 },
+                { statement: 'var a;a=1;var b;const c=1;', errorCount: 2 },
+                { statement: 'var a;/*block comment*/a=1;/*block comment2*/var b;const c = 1;', errorCount: 2 },
+                { statement: 'var a;function b(){var c;c=1;var d;var e;}', errorCount: 2 },
+                { statement: 'var a;for(var count=0;count<10;count++){var c;}', errorCount: 2 },
+                { statement: 'var a;var count;while(count<10){var b=1;}const c=1;', errorCount: 2 },
+                { statement: 'var a;var count;do{var b=1;}while(count<10);const c=1;', errorCount: 2 },
+                { statement: 'var a;for(var count=0; count<10;count++){var c;}var d;', errorCount: 3 },
+                { statement: 'var a; for (var count = 0; count < 10; count++) { var c; } var d;', errorCount: 3 },
+                { statement: 'var a;\nfor (var count = 0; count < 10; count++) {\nvar c;\n}\nvar d;', errorCount: 3 }
+            ];
+
+            scenarios.forEach(function(scenario) {
+                describe('statement' + JSON.stringify(scenario.statement), function() {
+                    it('should return ' + scenario.errorCount + ' error' + (scenario.errorCount === 1 ? '' : 's'),
```

Comment:
Not really a fan of that kind of declaration, i understand it is much easier to define, but it is much harder to support or isolate.

Not saying you should redo it, but want to hear some opinions about it.